sql/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectService.scala

```diff
@@ -69,6 +70,8 @@ class SparkConnectService(debug: Boolean) extends AsyncService with BindableServ
   override def executePlan(
       request: proto.ExecutePlanRequest,
       responseObserver: StreamObserver[proto.ExecutePlanResponse]): Unit = {
+    ConnectCommon.CONNECT_LOCAL_AUTH_TOKEN.foreach(k =>
+      assert(k == request.getUserContext.getLocalAuthToken))
```

Comment:
A couple of things. Please use a ServerSide Interceptor. This creates a narrow waist for authentication. This approach requires forces us to add authentication for each RPC, this is a problem when we add new RPCs.

Also do not use asserts. Asserts can be elided. Throw a GRPC Exception with status code ``UNAUTHENTICATED`.

Reply:
For the client side we can use `CallCredentials`. See the `SparkConnectClient.AccessTokenCallCredentials` for an example.